### PR TITLE
chore: add new integration tests scripts

### PR DESCRIPTION
**Pull request by davelsan**
## Description

This PR is a re-write of the integration tests command-line interface to only use a few bash scripts to generate the environment and run the tests.

The scripts are written with readability in mind, to facilitate other developers to make changes as the environment requirements evolve, or new features and fixes are requested.

## Changes
- Add a new suite of shell scripts that emulate what a human admin would do to run the tests.
- Remove the `zendro-env` dependency.
- Bump `package.json` to version 0.5.5.
- Add a simple command-line interface that exposes the original options to run the tests.
- Add a formatted man page with descriptions and examples of how to use the tests.
- Update the relevant section in the repository `README.md` file.

## Usage

```bash
# Use the already existing command to run the integration tests via npm
npm run test-integration [-- OPTIONS ]

# For more information about the available options, and some examples
npm run test-integration -- -h
```

## Spec

A (slightly outdated) document outlining the design decisions made by @asishallab can be found in the [gitbook workspace](https://app.gitbook.com/@zendro-dev/s/specifications/zendro-integration-tests) assigned to this PR.

### CODE REVIEW

File: test/testenv_generate_code.sh [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+  "${GRAPHQL_SERVER_1}/migrateDbAndStartServer.sh" \
+  "${TEST_DIR}/patches/migrateDbAndStartServer1.patch"
+
+patch \
+  "${GRAPHQL_SERVER_2}/migrateDbAndStartServer.sh" \
+  "${TEST_DIR}/patches/migrateDbAndStartServer2.patch"
+
+patch \
+  "${SPA_SERVER_1}/src/acl_rules.js" \
+  "${TEST_DIR}/patches/acl_rules.js.patch"
```
Comment (coeit) [inline test/testenv_generate_code.sh:110]: move to separate `testenv_sync.sh` script

**Review by coeit — CHANGES_REQUESTED** [review]

File: package.json [hunk 2]
```diff
@@ -28,14 +27,14 @@
     "js-beautify": "^1.6.11",
     "list-dir": "^1.2.0",
     "lodash": "^4.17.5",
+    "plotly.js": "^1.58.1",
```
[the remark this answers is not in the document]
Comment (davelsan) [reply package.json]: Fixed. This dependency was introduced by mistake.

File: test/testenv_generate_code.sh (hunk 1 above)
Comment (davelsan) [reply test/testenv_generate_code.sh]: Done.

**Comment by davelsan** [comment]
I'm thinking that the most glaring DRY violations, like "heading" debug output logs or the code to source testenv_constants.sh could be moved to helpers provided by testenv_constants.sh itself?

**Review by coeit — APPROVED** [review]

File: package.json [hunk 3]
```diff
@@ -11,8 +11,7 @@
     "single-page-app-codegen": "index.js"
   },
   "scripts": {
-    "test-env": "zenv",
-    "test-integration": "zenv integration",
+    "test-integration": "bash ./test/testenv_cli.sh",
     "test-integration-b": "bash ./test/previous/test/integration-test.sh"
```
Comment (framirez07) [inline package.json:15]: "test-integration-b" can be removed as well as the "/test/previous" dir.

File: test/testenv_cli.sh [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+  bash "${TEST_DIR}/testenv_docker_up.sh"
+  exit 0
+fi
+
+
+# RUN INTEGRATION TESTS
+# 1. Run the integration tests
+# 2. Perform a full cleanup (optionally disabled)
+if [[ $OPT_RUN_TESTS == "true" ]]; then
+  mocha "${TEST_DIR}/integration-tests-mocha.js"
```
Comment (framirez07) [inline test/testenv_cli.sh:112]:
```suggestion
  bash "${TEST_DIR}/testenv_docker_up.sh"
  mocha "${TEST_DIR}/integration-tests-mocha.js"
```

Comment (framirez07) [reply test/testenv_cli.sh:112]: This step regenerates the volumes, which is necessary for some of the tests to run correctly.

` bash "${TEST_DIR}/testenv_docker_up.sh"`

**Review by framirez07 — APPROVED** [review]
Ok. Just please see the two new comments.

File: package.json (hunk 3 above)
Comment (davelsan) [reply package.json:15]: Done.

File: test/testenv_cli.sh (hunk 4 above)
Comment (davelsan) [reply test/testenv_cli.sh:112]: Okay @framirez07, I have added this functionality.

**Pull request merged**